mlflow/gensim.py

```diff
@@ -0,0 +1,261 @@
+                        train = df.drop_column("target_label")
+                        signature = infer_signature(train, model.predict(train))
+    :param signature: (Experimental) :py:class:`ModelSignature <mlflow.models.ModelSignature>`
+                      describes model input and output :py:class:`Schema <mlflow.types.Schema>`.
+                      The model signature can be :py:func:`inferred <mlflow.models.infer_signature>`
+                      from datasets with valid model input (e.g. the training dataset with target
+                      column omitted) and valid model output (e.g. model predictions generated on
+                      the training dataset), for example:
+                      .. code-block:: python
+                        from mlflow.models.signature import infer_signature
+                        train = df.drop_column("target_label")
+                        predictions = ... # compute model predictions
+                        signature = infer_signature(train, predictions)
+    :param input_example: (Experimental) Input example provides one or several instances of valid
+                          model input. The example can be used as a hint of what data to feed the
+                          model. The given example will be converted to a Pandas DataFrame and then
+                          serialized to json using the Pandas split-oriented format. Bytes are
+                          base64-encoded.
+    :param await_registration_for: Number of seconds to wait for the model version to finish
+                            being created and is in ``READY`` status. By default, the function
+                            waits for five minutes. Specify 0 or None to skip waiting.
+    """
+    Model.log(
+        artifact_path=artifact_path,
+        flavor=mlflow.gensim,
+        registered_model_name=registered_model_name,
+        gensim_model=gensim_model,
+        conda_env=conda_env,
+        signature=signature,
+        input_example=input_example,
+        await_registration_for=await_registration_for,
+        **kwargs
+    )
+
+
+def _load_model_from_local_file(path: str) -> Any:
+    """
+    Load a Gensim model saved as an MLflow artifact on the local file system.
+    :param path: Local filesystem path to the MLflow Model with the ``gensim`` flavor.
+    """
+    with open(path, "rb") as f:
+        return pickle.load(f)
+
+
+def _load_pyfunc(path: str) -> Any:
+    """
+    Load PyFunc implementation. Called by ``pyfunc.load_pyfunc``.
+    :param path: Local filesystem path to the MLflow Model with the ``sklearn`` flavor.
+    """
+    return _load_model_from_local_file(path)
+
+
+def _save_model(gensim_model, output_path: str) -> None:
+    """
+    :param gensim_model: The Gensim model to serialize.
+    :param output_path: The file path to which to write the serialized model.
+    """
+    with open(output_path, "wb") as out:
+        pickle.dump(gensim_model, out)
```

Comment:
It looks like gensim models have a `save` method. I'm wondering which one (pickle vs save) we should use:

```python
from gensim.test.utils import common_texts
from gensim.models import Word2Vec
>>>
model = Word2Vec(sentences=common_texts, vector_size=100, window=5, min_count=1, workers=4)
model.save("word2vec.model")
```


https://radimrehurek.com/gensim/models/word2vec.html#usage-examples